staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go

```diff
@@ -430,10 +430,47 @@ func AddDryRunFlag(cmd *cobra.Command) {
 	cmd.Flags().Lookup("dry-run").NoOptDefVal = "unchanged"
 }
 
+const (
+	fieldManagerClientSideApply = "kubectl-client-side-apply"
+	// The default server-side apply field manager is `kubectl`
+	// instead of a field manager like `kubectl-server-side-apply`
+	// for backward compatibility to not conflict with old versions
+	// of kubectl server-side apply where `kubectl` has already been the field manager.
+	fieldManagerServerSideApply = "kubectl"
+)
+
+// GetApplyFieldManagerFlag gets the field manager for kubectl apply
+// if it is not set.
+//
+// The default field manager is not `kubectl-apply` to distinguish between
+// client-side and server-side apply.
+func GetApplyFieldManagerFlag(cmd *cobra.Command, serverSide bool) string {
+	// The field manager flag was set
+	if cmd.Flag("field-manager").Changed {
+		return GetFlagString(cmd, "field-manager")
+	}
+
+	if serverSide {
+		return fieldManagerServerSideApply
+	}
+
+	return fieldManagerClientSideApply
+}
+
+func AddFieldManagerFlagVar(cmd *cobra.Command, p *string, defaultFieldManager string) {
+	cmd.Flags().StringVar(p, "field-manager", defaultFieldManager, "Name of the manager used to track field ownership.")
+}
+
+func addFieldManagerFlag(cmd *cobra.Command, defaultFieldManager string) *string {
+	p := new(string)
+	AddFieldManagerFlagVar(cmd, p, defaultFieldManager)
+	return p
+}
+
 func AddServerSideApplyFlags(cmd *cobra.Command) {
 	cmd.Flags().Bool("server-side", false, "If true, apply runs in the server instead of the client.")
 	cmd.Flags().Bool("force-conflicts", false, "If true, server-side apply will force the changes against conflicts.")
-	cmd.Flags().String("field-manager", "kubectl", "Name of the manager used to track field ownership.")
+	addFieldManagerFlag(cmd, fieldManagerClientSideApply)
```

Comment:
I'm not super excited about the changes in this file. Can we just ermove the FieldManager flag from being registered in that method, since clearly it's done in many places now.

I'm also not sure if `GetApplyFieldManagerFlag` belongs here, I'd rather see it in `apply.go`.

I don't think I've thought through everything so I might be wrong.

Reply:
Cleaned up!